File: scope-intelligence-toolkit/examples/sample-repo/src/frontend/apiClient.js
const API_BASE = process.env.API_BASE || "http://localhost:3000";
const API_TIMEOUT = process.env.API_TIMEOUT_MS || "5000";

export async function postJson(url, payload) {
    const res = await fetch(`${API_BASE}${url}`, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(payload),
    });
    return res.json();
}

export async function getJson(url) {
    const res = await fetch(`${API_BASE}${url}`);
    return res.json();
}

File: examples/sample-repo/src/frontend/api_client.py
"""Frontend API client."""
import os

API_BASE = os.environ.get("API_BASE", "http://localhost:8000")
API_TIMEOUT_MS = int(os.environ.get("API_TIMEOUT_MS", "3000"))


def post_login(username: str, password: str) -> dict:
    import urllib.request, json
    body = json.dumps({"username": username, "password": password}).encode()
    req = urllib.request.Request(f"{API_BASE}/api/login", data=body,
                                 headers={"Content-Type": "application/json"})
    with urllib.request.urlopen(req, timeout=API_TIMEOUT_MS / 1000) as r:
        return json.loads(r.read())


def post_charge(token: str, amount: float) -> dict:
    import urllib.request, json
    body = json.dumps({"token": token, "amount": amount}).encode()
    req = urllib.request.Request(f"{API_BASE}/api/billing/charge", data=body,
                                 headers={"Content-Type": "application/json"})
    with urllib.request.urlopen(req, timeout=API_TIMEOUT_MS / 1000) as r:
        return json.loads(r.read())
